src/predicates/propEq.spec.js

```diff
@@ -0,0 +1,52 @@
+const test = require('tape')
+const helpers = require('../test/helpers')
+
+const bindFunc = helpers.bindFunc
+
+const isFunction = require('../core/isFunction')
+const unit = require('../core/_unit')
+
+const propEq = require('./propEq')
+
+test('propEq function', t => {
+  const f = bindFunc(propEq)
+
+  t.ok(isFunction(propEq), 'is a function')
+
+  const err1 = /propEq: Number or String required for first argument/
+  const err3 = /propEq: Object required for third argument/
+  const key = 'something'
+  const keyN = 1
+  const val = 'thirty-six'
+  const obj = { [key]: val }
+
+  t.throws(f(undefined, val, {}), err1, 'throws with undefined in first argument')
+  t.throws(f(null, val, {}), err1, 'throws with null in first argument')
+  t.throws(f(false, val, {}), err1, 'throws with false in first argument')
+  t.throws(f(true, val, {}), err1, 'throws with true number in first argument')
+  t.throws(f({}, val, {}), err1, 'throws with object in first argument')
+  t.throws(f([], val, {}), err1, 'throws with array in first argument')
+  t.throws(f(unit, val, {}), err1, 'throws with function in first argument')
+  t.ok(f(key, val, obj), 'accepts a string in the first argument')
+  t.ok(f(keyN, val, obj), 'accepts a number in the first argument')
+
+  t.throws(f(key, val, undefined), err3, 'throws with undefined in third argument')
+  t.throws(f(key, val, null), err3, 'throws with null in third argument')
+  t.throws(f(key, val, false), err3, 'throws with false in third argument')
+  t.throws(f(key, val, true), err3, 'throws with true number in third argument')
+  t.throws(f(key, val, 5), err3, 'throws with number in third argument')
+  t.throws(f(key, val, 'hello'), err3, 'throws with string in third argument')
+  t.throws(f(key, val, []), err3, 'throws with array in third argument')
+  t.throws(f(key, val, unit), err3, 'throws with function in third argument')
+  t.ok(f(key, val, {}), 'accepts an object in the third argument')
+
+  t.equals(propEq(key, val, { [key]: undefined }), false, 'returns false when value does not exist on object')
+  t.equals(propEq(key, '5', { [key]: 5 }), false, 'returns false for truthy string')
+  t.equals(propEq(key, 5, { [key]: '5' }), false, 'returns false for truthy number')
+
+  t.equals(propEq(key, val, obj), true, 'returns true when value exists on object')
```

Comment:
So the reason this is failing is due to compilation issues, when `[key]: ` is used above, buble creates an `obj` to which it uses in a sequence expression so when it checks to see if the value is true here, it is not. it is actually '5'.

So to get around this should probably just loose the obj assignment and use a literal in these cases like you did above:
```js
t.equals(propEq('a', 232, { a: 232 }), true, 'returns true when value exists on object')
```

That will clear up the error. Or you can pick a different name then `obj` if you wanna keep it like this

Reply:
Wow, that's pretty interesting. I'll keep that in mind.